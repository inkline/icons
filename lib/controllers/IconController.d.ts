import { INode as Svg } from "svgson";
export interface IIconController {
    icons: {
        [key: string]: Svg;
    };
    add(name: string, icon: Svg): void;
    addMultiple(icons: {
        [key: string]: Svg;
    }): void;
}
export declare const IconController: IIconController;
export default IconController;
