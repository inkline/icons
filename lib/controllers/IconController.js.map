{"version":3,"file":"IconController.js","sourceRoot":"","sources":["../../src/controllers/IconController.ts"],"names":[],"mappings":"AAQA,MAAM,CAAC,MAAM,cAAc,GAAoB;IAC3C,KAAK,EAAE,EAAE;IACT,GAAG,CAAC,IAAI,EAAE,IAAI;QACV,cAAc,CAAC,KAAK,CAAC,IAAI,CAAC,GAAG,IAAI,CAAC;IACtC,CAAC;IACD,WAAW,CAAC,KAAK;QACb,MAAM,CAAC,IAAI,CAAC,KAAK,CAAC,CAAC,OAAO,CAAC,CAAC,QAAQ,EAAE,EAAE;YACpC,cAAc,CAAC,KAAK,CAAC,QAAQ,CAAC,GAAG,KAAK,CAAC,QAAQ,CAAC,CAAC;QACrD,CAAC,CAAC,CAAC;IACP,CAAC;CACJ,CAAC;AAEF,eAAe,cAAc,CAAC"}