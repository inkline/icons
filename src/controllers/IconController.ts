import { INode as Svg } from "svgson";

export interface IIconController {
    icons: { [key: string]: Svg };
    add(name: string, icon: Svg): void;
    addMultiple(icons: { [key: string]: Svg }): void;
}

export const IconController: IIconController = {
    icons: {},
    add(name, icon) {
        IconController.icons[name] = icon;
    },
    addMultiple(icons) {
        Object.keys(icons).forEach((iconName) => {
            IconController.icons[iconName] = icons[iconName];
        });
    }
};

export default IconController;
