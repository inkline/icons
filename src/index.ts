import * as inklineIconsPack from '@inkline/icons/packs/inkline';

export { inklineIconsPack };
export * from '@inkline/icons/config';
export * from '@inkline/icons/components';
export * from '@inkline/icons/controllers';
export * from '@inkline/icons/helpers';
export * from '@inkline/icons/plugin';
export * from '@inkline/icons/packs';
