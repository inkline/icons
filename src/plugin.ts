import { Plugin } from 'vue';
import { IconController } from "./controllers";
import { Svg } from './types';
import { IIcon } from "./components";

export type InklineIconsPlugin = Plugin & {
    add(icons: { [key: string]: Svg }): void;
};

export interface PluginOptions {
    registerComponent?: boolean;
}

const defaultOptions: PluginOptions = {
    registerComponent: true
};

export const InklineIcons: InklineIconsPlugin = {
    add(icons) {
        IconController.addMultiple(icons);
    },
    install(app, options= {}) {
        options = {
            ...defaultOptions,
            ...options
        };

        app.provide('inklineIcons', IconController);

        if (options.registerComponent) {
            app.component(IIcon.name, IIcon);
        }
    }
};

export default InklineIcons;
