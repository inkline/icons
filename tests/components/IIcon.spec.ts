import { shallowMount } from '@vue/test-utils';
import { IIcon } from '../../src/components';
import { inkCheck } from "../../src/packs/inkline";
import { IconController } from "../../src/controllers";

IconController.add('icon', inkCheck);

describe('components', () => {
   describe('IIcon', () => {
       it('should render correctly', () => {
           const wrapper = shallowMount(IIcon, {
               global: {
                   provide: {
                       inklineIcons: IconController
                   }
               },
               props: {
                   name: 'icon'
               }
           });

           expect(wrapper.html()).toMatchSnapshot();
       });

       it('should not render if specified icon does not exist', () => {
           expect(() => shallowMount(IIcon, {
               global: {
                   provide: {
                       inklineIcons: IconController
                   }
               },
               props: {
                   name: 'doesnotexist'
               }
           })).toThrowError();
       });
   });
});
