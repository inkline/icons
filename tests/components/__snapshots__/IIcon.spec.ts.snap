// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components IIcon should render correctly 1`] = `
"<svg class=\\"inkline-icon\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"28\\" height=\\"28\\" viewBox=\\"0 0 28 28\\" fill=\\"currentColor\\">
  <title>check</title>
  <path d=\\"M23.625 3.5l-13.125 13.125-6.125-6.125-4.375 4.375 10.5 10.5 17.5-17.5z\\"></path>
</svg>"
`;
